tests/integration/select2-methods.js

```diff
@@ -137,3 +137,70 @@ test('multiple value matches the jquery value', function (assert) {
     'The values should match the jquery values'
   );
 });
+
+test('multiple selection and clearing of grouped options', function (assert) {
+  var container = new MockContainer();
+  var $container = $('<div></div>');
+  var $select = $('<select></select>');
+  $container.append($select);
+
+  var data = [{
+    text: 'Group 1',
+    children: [{
+      id: 1,
+      text: 'Option 1.1'
+    }, {
+      id: 2,
+      text: 'Option 1.2'
+    }]
+  }, {
+    text: 'Group 2',
+    children: [{
+      id: 3,
+      text: 'Option 2.1'
+    }, {
+      id: 4,
+      text: 'Option 2.2'
+    }]
+  }];
+
+  var select = new Select2($select, {
+    multiple: true,
+    data: data
+  });
+  select.render();
+  select.selection.bind(container, $container);
+
+  $select.val(['3', '1']);
+  $select.trigger('change');
+
+  assert.equal(
+    $select.find(':selected').length,
+    2,
+    'Two items should be selected'
+  );
+
+  var $remove = null;
+
+  // Remove the first item
+  $remove = $container.find('.select2-selection__choice__remove');
+  $remove.trigger('click');
+
+  // Open the dropdown menu
+  $container.find('.select2-selection').trigger('click');
+  var $selections = $('.select2-results__option[aria-selected=true]');
+
+  assert.equal(
+    $selections.length,
+    1,
+    'One item should remain selected'
+  );
+
+  // Remove the second selection by clicking on the item in the dropdown
+  $selections.first().trigger('click');
```

Comment:
The idea with this line is to simulate the user clicking on the dropdown menu item, so that we can check that the selection is removed afterwards. It doesn't appear to invoke a relevant event handler though.

Reply:
Changing the event from `click` to `mouseup` -- to correspond with [the event handler here](https://github.com/openculinary/select2/blob/db929cb6b835b30a31af72932335179fafb61f8d/src/js/select2/results.js#L451) seems to make a difference.  It causes the `dom-changes.js` integration test to break, though:

```
>> select2(val) - multiple selection and clearing of grouped options
>> Message: Too many calls to the `assert.async` callback
>> Actual: null
>> Expected: undefined
>> http://localhost:9999/tests/integration/dom-changes.js:284:14
```